Data/Functor/Foldable.hs

```diff
@@ -211,13 +213,18 @@ class Functor (Base t) => Corecursive t where
 
   -- | A generalized postpromorphism
   gpostpro
-    :: (Recursive t, Monad m)
-    => (forall b. m (Base t b) -> Base t (m b)) -- distributive law
-    -> (forall c. Base t c -> Base t c)         -- natural transformation
-    -> (a -> Base t (m a))                      -- a (Base t)-m-coalgebra
-    -> a                                        -- seed
```

Comment:
Sadly, I had to remove the comments because I wasn't sure what to call the `a -> Base t s`; it's certainly not a "`(Base t)`-`m`-coalgebra" anymore. I guess I could have kept the other comments...